frontend/src/components/hooks/groupedQueryParams.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  nullToUndefined,
+  useExcludeCategoriesParam,
+  useExcludeCollectionsParam,
+  useExcludeFilesParam,
+  useFolioParam,
+  useIncludeCategoriesParam,
+  useIncludeCollectionsParam,
+  useIncludeFilesParam,
+  useLanguageParam,
+  useLanguagesParam,
+  useParLengthParam,
+  useScoreParam,
+} from "@components/hooks/params";
```

Comment:
I see we are using `nuqs` v1, while v2 is out now. v2 is boasting nextjs 14 & 15 support, and we are on 14. Maybe v1 also works on v14 of next then? :) Still, it's probably best to upgrade if it's [not too much hassle](https://nuqs.47ng.com/blog/nuqs-2#breaking-changes--migration).

Reply:
Yes,  just today I also noticed v2 had been launched. I've already bumped it up on DharmaMitra and it was very straightforward. Will do that here.